addons/common/functions/fnc_registerItemReplacement.sqf

```diff
@@ -54,21 +60,18 @@ private _fnc_replaceItems = {
             };
         } forEach GVAR(inheritedReplacements);
 
-        // Skip lookup for all following items of this class
-        private _count = 1;
-        while {_newItems#(_i + 1) == _item} do { // (i+1) can be out of bounds, but should fail safely
-            _count = _count + 1;
-            _i = _i + 1;
-        };
-
         // Replace all items of current class in list
         if (_replacements isNotEqualTo []) then {
             TRACE_3("replace",_item,_count,_replacements);
             _unit removeItems _item;
 
-            for "_j" from 1 to _count do {
-                { [_unit, _x] call FUNC(addToInventory) } forEach _replacements;
-            };
+            {
+                if (_x == 0) then {continue};
+                private _container = ["uniform", "vest", "backpack"] select _forEachIndex;
+                for "_j" from 1 to _x do {
+                    { [_unit, _x, _container] call FUNC(addToInventory) } forEach _replacements;
+                }
+            } forEach _containerCount;
```

Comment:
This would be a little bit safer in case we don't have room in the original container
```
{
    if (_x == 0) then {continue};
    private _container = ["uniform", "vest", "backpack"] select _forEachIndex;
    for "_j" from 1 to _x do {
        { 
            if ([_unit, _x, 1, _container == "uniform", _container == "vest", _container == "backpack"] call CBA_fnc_canAddItem) then {
                [_unit, _x, _container] call FUNC(addToInventory)  // add to specific container
            } else {
                [_unit, _x, ""] call FUNC(addToInventory) // no room, add anywhere
            }
        } forEach _replacements;
    }
} forEach _containerCount;
```
e.g.
```
for "_i" from 0 to 9 do { ["ACE_RangeCard", "ACE_salineIV"] call ace_common_fnc_registerItemReplacement; };
player addItemToUniform "ACE_RangeCard";
```